src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -357,6 +357,7 @@ public override SyntaxNode AsPublicInterfaceImplementation(SyntaxNode declaratio
             return PreserveTrivia(declaration, d =>
             {
                 d = AsImplementation(d, Accessibility.Public);
+                d = WithoutConstraints(d);
```

Comment:
This is in 'AsPublicInterfaceImplementation`.  Do you want to remove constraints in this case?